### PR TITLE
fix multi-line lambda autocorrect whitespace issue

### PR DESCRIPTION
**Pull request by zvkemp**
Fixes an issue in which multi-line lambdas of the form

```ruby
->(x){
  x
}
```

...would be auto-corrected to:

```ruby
lambdado |x|
  x
end
```

This was the result of a multi-step autocorrect wherein the curly brackets were replaced by Cop::Style::Blocks, and then the Cop::Style::Lambda would convert the arguments to block-style without correcting for whitespace.

These changes ensures a space is added before `do ... end` blocks in general (if necessary), and fixes the whitespace issues with multi-line lambda arguments.

```
rubocop -V
0.29.1 (using Parser 2.2.0.3, running on ruby 2.1.5 x86_64-darwin14.0)
```

### CODE REVIEW

File: lib/rubocop/cop/style/lambda.rb [hunk 1]
```diff
@@ -22,11 +22,11 @@ def on_block(node)
 
           return unless block_method == TARGET
           selector = block_method.loc.selector.source
-          lambda_length = lambda_length(node)
+          length = lambda_length(node)
```
Comment (bbatsov) [inline lib/rubocop/cop/style/lambda.rb]: Why did you rename this?

File: lib/rubocop/cop/style/lambda.rb [hunk 2]
```diff
@@ -62,8 +62,14 @@ def autocorrect(node)
 
         def autocorrect_new_to_old(corrector, node)
           block_method, args = *node
+          if add_whitespace?(block_method, args, node)
+            corrector.insert_before(node.loc.begin, ' ')
+          end
           corrector.replace(block_method.loc.expression, 'lambda')
-          return if args.children.empty?
+          if args.children.empty?
```
Comment (bbatsov) [inline lib/rubocop/cop/style/lambda.rb]: This `if` looks kind of ugly because of the next `if` and the `return`. I'd suggest simplifying it a bit.

File: lib/rubocop/cop/style/lambda.rb [hunk 3]
```diff
@@ -62,8 +62,14 @@ def autocorrect(node)
 
         def autocorrect_new_to_old(corrector, node)
           block_method, args = *node
+          if add_whitespace?(block_method, args, node)
```
Comment (bbatsov) [inline lib/rubocop/cop/style/lambda.rb]: Some comment here would be helpful I think.

**Comment by bbatsov** [comment]
Please, capitalize the commit message.

File: lib/rubocop/cop/style/lambda.rb (hunk 1 above)
Comment (zvkemp) [inline lib/rubocop/cop/style/lambda.rb]: To avoid colliding with the `lambda_length` method in that context.

**Comment by zvkemp** [comment]
Thanks @bbatsov. I've incorporated your suggestions into the branch.

**Comment by bbatsov** [comment]
:+1: Thanks!
